tests/datavolume_test.go

```diff
@@ -811,6 +811,108 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 		})
 	})
 
+	Describe("[rfe_id:1115][crit:high][vendor:cnv-qe@redhat.com][level:component][test] CDI Import from HTTP/S3", func() {
+
+		var (
+			dataVolume              *cdiv1.DataVolume
+			err                     error
+			originalImageName       = "cirros-qcow2.img"
+			testImageName           = "cirros-qcow2-1990.img"
+			tinyCoreIsoRateLimitURL = fmt.Sprintf("http://cdi-file-host.%s:82/cirros-qcow2-1990.img", f.CdiInstallNs)
+		)
+
+		BeforeEach(func() {
+			By("Prepare the file")
+			fileHostPod, err := utils.FindPodByPrefix(f.K8sClient, f.CdiInstallNs, utils.FileHostName, "name="+utils.FileHostName)
+			_, _, err = f.ExecCommandInContainerWithFullOutput(fileHostPod.Namespace, fileHostPod.Name, "http",
+				"/bin/sh",
+				"-c",
+				"cp /tmp/shared/images/"+originalImageName+" /tmp/shared/images/"+testImageName)
+			Expect(err).To(BeNil())
+		})
+
+		AfterEach(func() {
+			By("Delete DV")
+			err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Cleanup the file")
```

Comment:
Do we want to check if the original file is there, and if not restore it. It is possible the test failed after you removed the original file, and before you restored it. Then this after each will remove the backup as well, and we will have lost the file completely.

Reply:
The thing is, I am never removing the original, I am doing test on new `testImageName` file. So cleanup is simpler, and in case of any errors we do not loose original we just leave a little bit of garbage as `testImageName` (but this is handled in afterEach, so should not happen)

Reply:
Good point, I read it wrong, I thought you removed the original and then put it back, but if you are making a copy and then removing it, that works.